File: backend/internal/model/scrapetask.go
package model

import "time"

// ScrapeTask 刮削计划任务
type ScrapeTask struct {
	ID          uint       `json:"id" gorm:"primaryKey"`
	Name        string     `json:"name" gorm:"type:text"`           // 任务名称
	LibraryID   string     `json:"library_id" gorm:"type:text;index"` // 媒体库 ID
	LibraryName string     `json:"library_name" gorm:"type:text"`   // 媒体库名称（冗余存储）
	Interval    int        `json:"interval" gorm:"default:60"`      // 扫描频率（分钟）
	Enabled     bool       `json:"enabled" gorm:"default:true"`    // 是否启用
	LastRunAt   *time.Time `json:"last_run_at"`                     // 上次执行时间
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
}

// TableName 指定表名
func (ScrapeTask) TableName() string {
	return "scrape_tasks"
}

File: frontend/src/api/scrapetask.ts
import request from './request'

// 刮削计划任务
export interface ScrapeTask {
  id: number
  name: string
  library_id: string
  library_name: string
  interval: number // 扫描频率（分钟）
  enabled: boolean
  last_run_at: string | null
  is_running?: boolean // 是否正在运行
  created_at: string
  updated_at: string
}

// 获取刮削计划任务列表
export const getScrapeTasks = () =>
  request.get<ScrapeTask[]>('/scrapetask/list')

// 创建刮削计划任务
export const createScrapeTask = (data: { name: string; library_id: string; library_name: string; interval: number; enabled: boolean }) =>
  request.post<ScrapeTask>('/scrapetask/create', data)

// 更新刮削计划任务
export const updateScrapeTask = (data: { id: number; name: string; library_id: string; library_name: string; interval: number; enabled: boolean }) =>
  request.post<ScrapeTask>('/scrapetask/update', data)

// 删除刮削计划任务
export const deleteScrapeTask = (id: number) =>
  request.delete(`/scrapetask/${id}`)

// 立即执行刮削计划任务
export const runScrapeTask = (id: number) =>
  request.post<{ status: string; message: string }>(`/scrapetask/run/${id}`)
